### ISSUE TITLE
require-jsdoc false positive on first property after static field within class

### ISSUE THREAD

**Issue by designbystephen** [babel/babel-eslint#522]
ILink to my original issue I submitted on the ESLint repo:
https://github.com/eslint/eslint/issues/9356

```js

/**
 * Employee component
 */
export class Employee extends React.Component {
  /** @type {Object} component prop types */
  static propTypes = {
    match: PropTypes.object.isRequired,
    title: PropTypes.string.isRequired,
    children: PropTypes.any.isRequired
  }

  /**
   * Construct component
   * @param {*} props - component properties
   */
  constructor(props){
    super(props);

    this.name = null;
  }

  /**
   * Pre-render
   */
  componentWillMount(){
    this.name = this.fetchName();
  }
}

```

In the above example I am getting, what I believe to be, a false positive on the first property/mehod defined after a property declaration like: `static propTypes = {}`.  If I change the declaration to ` static get propTypes(){}` then it no longer highlights as an error.  Both are proper ways to declare a static property, as far as I know, so I would expect to not see any lint errors with either declaration. 

Thanks for your hard work and as always any help would be greatly appreciated.

**Comment by kaicataldo** [comment]
Thank you for the issue. Now that [`@babel/eslint-parser` has been released](https://babeljs.io/blog/2020/07/13/the-state-of-babel-eslint), we are making this repository read-only. If this is a change you would still like to advocate for, please reopen this in the [babel/babel monorepo](https://github.com/babel/babel).
